### PR TITLE
Apply errorprone to all java projects

### PR DESCRIPTION
**Pull request by epeee**
I thought it might be a good idea to apply the errorprone plugin to all projects using the java plugin.
Furthermore, I think it is a good idea to specify the version of errorprone in the gradle file instead of always using the latest one (since this might break the build e.g. due to new checks).

@TimvdLippe, @mockitoguy what do you think?

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ buildscript {
 
 plugins {
     id 'com.gradle.build-scan' version '1.12.1'
-    id 'net.ltgt.errorprone' version '0.0.13'
+    id 'net.ltgt.errorprone' version '0.0.13' apply false
```
Comment (TimvdLippe) [inline build.gradle]: Why not remove this line and put it on line 53 instead?

File: build.gradle [hunk 2]
```diff
@@ -45,10 +45,16 @@ apply from: 'gradle/dependencies.gradle'
...
 
-allprojects {
+allprojects { proj ->
     repositories {
         jcenter()
     }
+    plugins.withId('java') {
+        proj.apply plugin: 'net.ltgt.errorprone'
+        proj.dependencies {
+            errorprone 'com.google.errorprone:error_prone_core:2.2.0'
```
Comment (TimvdLippe) [inline build.gradle]: Let's extract the version number into the `dependencies.gradle` file

**Comment by szpak** [comment]
There seems to be a problem with errorprone and Java 10 - Travis in master [fails](https://travis-ci.org/mockito/mockito/jobs/361255753#L853-L865).

**Comment by TimvdLippe** [comment]
Errorprone on JDK 10 works on `release/2.x` here though 🤔 https://travis-ci.org/mockito/mockito/builds/361158556

**Comment by epeee** [comment]
thx, I will have a look at this one.

**Pull request merged**